tools/modulelint.py

```diff
@@ -46,8 +46,30 @@ def setUp(self):
     def testDockerFromBaseruntime(self):
         self.assertTrue(self.dp.check_baseruntime())
 
-    def testDockerRunMicrodnf(self):
-        self.assertTrue(self.dp.check_microdnf())
+    def testDockerNodocs(self):
+        self.start()
+        installed_pkgs = self.run("rpm -qa --qf '%{{NAME}}\n'", ignore_status=True).stdout
```

Comment:
why do you need to go through all packages, check their files... why not to go through all docs and check files directly?...

Remember that each RPM query slows down test.

Reply:
This tests verifies if package is really installed on the system.

Reply:
FYI
if I run meta-test-family/examples/testing-module with this PR=20 I can see fail in this new tests within nspawn part:
try:
MODULE=nspawn CONFIG=./fullconfig.yaml avocado run --filter-by-tags=-WIP copyTest.py escapingTest.py microdnfinside.py PureAvocadoTest.py rpmvalidator.py shellTest.sh simpleTest.py skipTest.py ../../tools/modulelint/check_compose.py ../../tools/modulelint/modulelint.py ../../tools/modulelint/rpmvalidation.py

in meta-test-family/examples/testing-module
 LOG:

:: [  BEGIN   ] :: Running 'make check-rpm'
make[1]: Entering directory '/tmp/tmp.i11ypNhE6c/meta-test-family/examples/testing-module'
MODULE=nspawn mtf-env-set
name of CHROOT directory:
/opt/chroot_bash_1503904476.961487
Disabling selinux
Disabling selinux
MODULE=nspawn CONFIG=./fullconfig.yaml avocado run --filter-by-tags=-WIP copyTest.py escapingTest.py microdnfinside.py PureAvocadoTest.py rpmvalidator.py shellTest.sh simpleTest.py skipTest.py ../../tools/modulelint/check_compose.py ../../tools/modulelint/modulelint.py ../../tools/modulelint/rpmvalidation.py
JOB ID     : 7d4fb166075064edaa68635c68814cfc94aef35a
JOB LOG    : /root/avocado/job-results/job-2017-08-28T03.14-7d4fb16/job.log
 (01/34) copyTest.py:CheckCopyFiles.testCopyThereAndBack:  PASS (45.34 s)
 (02/34) escapingTest.py:SanityCheckApostophes.testSimpleShell:  PASS (40.60 s)
 (03/34) escapingTest.py:SanityCheckApostophes.testApostrophesInside:  PASS (41.58 s)
 (04/34) escapingTest.py:SanityCheckApostophes.testMoreLevelApostrophes:  PASS (37.11 s)
 (05/34) escapingTest.py:SanityCheckApostophes.testStderr:  PASS (56.27 s)
 (06/34) microdnfinside.py:microDNFTest.testInstallMicroDNFEmpty:  PASS (46.75 s)
 (07/34) PureAvocadoTest.py:PureAvocadoTest.testInsideModule:  PASS (36.73 s)
 (08/34) PureAvocadoTest.py:PureAvocadoTest.testOnHost:  PASS (35.82 s)
 (09/34) rpmvalidator.py:ExampleRpmValidation.testPaths:  PASS (55.54 s)
 (10/34) shellTest.sh:  PASS (41.18 s)
 (11/34) simpleTest.py:simpleTests.testPath:  PASS (33.61 s)
 (12/34) simpleTest.py:simpleTests.testAssertIn:  PASS (36.65 s)
 (13/34) simpleTest.py:simpleTests.testInsideModule:  PASS (35.18 s)
 (14/34) simpleTest.py:simpleTests.testCommandOnHost:  PASS (34.86 s)
 (15/34) skipTest.py:SkipTest.testGccSkippedInsideTest:  CANCEL (34.29 s)
 (16/34) skipTest.py:SkipTest.testDecoratorNotSkippedForDefault:  SKIP
 (17/34) skipTest.py:SkipTest.testDecoratorSkip:  SKIP
 (18/34) ../../tools/modulelint/check_compose.py:ComposeTest.test_component_profile_installability:  PASS (36.80 s)
 (19/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testDockerFromBaseruntime:  PASS (1.04 s)
 (20/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testDockerNodocs:  ERROR (1.07 s)
 (21/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testDockerCleanAll:  ERROR (1.07 s)
 (22/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testArchitectureInEnvAndLabelExists:  PASS (1.04 s)
 (23/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testNameInEnvAndLabelExists:  PASS (1.04 s)
 (24/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testReleaseLabelExists:  PASS (1.05 s)
 (25/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testVersionLabelExists:  PASS (1.04 s)
 (26/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testComRedHatComponentLabelExists:  PASS (1.04 s)
 (27/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testIok8sDescriptionExists:  PASS (1.04 s)
 (28/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testIoOpenshiftExposeServicesExists:  PASS (1.04 s)
 (29/34) ../../tools/modulelint/modulelint.py:DockerfileLinter.testIoOpenShiftTagsExists:  PASS (1.04 s)
 (30/34) ../../tools/modulelint/modulelint.py:DockerLint.testBasic:  SKIP
 (31/34) ../../tools/modulelint/modulelint.py:DockerLint.testContainerIsRunning:  SKIP
 (32/34) ../../tools/modulelint/modulelint.py:DockerLint.testLabels:  SKIP
 (33/34) ../../tools/modulelint/modulelint.py:ModuleLintPackagesCheck.test:  PASS (34.92 s)
 (34/34) ../../tools/modulelint/rpmvalidation.py:rpmvalidation.test:  PASS (190.79 s)
RESULTS    : PASS 26 | ERROR 2 | FAIL 0 | SKIP 5 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB TIME   : 894.55 s
JOB HTML   : /root/avocado/job-results/job-2017-08-28T03.14-7d4fb16/results.html
make[1]: *** [Makefile:23: check-rpm] Error 1
make[1]: Leaving directory '/tmp/tmp.i11ypNhE6c/meta-test-family/examples/testing-module'
:: [[0;31m   FAIL   [0m] :: Command 'make check-rpm' (Expected 0, got 2)